src/core/ext/filters/client_channel/retry_filter.cc

```diff
@@ -120,2494 +84,59 @@
-      &call_attempt_->recv_message_ready_;
-}
-
-void RetryFilter::CallData::CallAttempt::BatchData::
-    AddRetriableRecvTrailingMetadataOp() {
-  call_attempt_->started_recv_trailing_metadata_ = true;
-  batch_.recv_trailing_metadata = true;
-  call_attempt_->recv_trailing_metadata_.Clear();
-  batch_.payload->recv_trailing_metadata.recv_trailing_metadata =
-      &call_attempt_->recv_trailing_metadata_;
-  batch_.payload->recv_trailing_metadata.collect_stats =
-      &call_attempt_->collect_stats_;
-  GRPC_CLOSURE_INIT(&call_attempt_->recv_trailing_metadata_ready_,
-                    RecvTrailingMetadataReady, this, grpc_schedule_on_exec_ctx);
-  batch_.payload->recv_trailing_metadata.recv_trailing_metadata_ready =
-      &call_attempt_->recv_trailing_metadata_ready_;
-}
-
-void RetryFilter::CallData::CallAttempt::BatchData::AddCancelStreamOp(
-    grpc_error_handle error) {
-  batch_.cancel_stream = true;
-  batch_.payload->cancel_stream.cancel_error = error;
-  // Override on_complete callback.
-  GRPC_CLOSURE_INIT(&on_complete_, OnCompleteForCancelOp, this, nullptr);
-}
-
-//
-// CallData vtable functions
-//
-
-grpc_error_handle RetryFilter::CallData::Init(
-    grpc_call_element* elem, const grpc_call_element_args* args) {
-  auto* chand = static_cast<RetryFilter*>(elem->channel_data);
-  new (elem->call_data) CallData(chand, *args);
-  if (GRPC_TRACE_FLAG_ENABLED(grpc_retry_trace)) {
-    gpr_log(GPR_INFO, "chand=%p calld=%p: created call", chand,
-            elem->call_data);
-  }
-  return absl::OkStatus();
-}
-
-void RetryFilter::CallData::Destroy(grpc_call_element* elem,
-                                    const grpc_call_final_info* /*final_info*/,
-                                    grpc_closure* then_schedule_closure) {
-  auto* calld = static_cast<CallData*>(elem->call_data);
-  // Save our ref to the CallStackDestructionBarrier until after our
-  // dtor is invoked.
-  RefCountedPtr<CallStackDestructionBarrier> call_stack_destruction_barrier =
-      std::move(calld->call_stack_destruction_barrier_);
-  calld->~CallData();
-  // Now set the callback in the CallStackDestructionBarrier object,
-  // right before we release our ref to it (implicitly upon returning).
-  // The callback will be invoked when the CallStackDestructionBarrier
-  // is destroyed.
-  call_stack_destruction_barrier->set_on_call_stack_destruction(
-      then_schedule_closure);
-}
-
-void RetryFilter::CallData::StartTransportStreamOpBatch(
-    grpc_call_element* elem, grpc_transport_stream_op_batch* batch) {
-  auto* calld = static_cast<CallData*>(elem->call_data);
-  calld->StartTransportStreamOpBatch(batch);
-}
-
-void RetryFilter::CallData::SetPollent(grpc_call_element* elem,
-                                       grpc_polling_entity* pollent) {
-  auto* calld = static_cast<CallData*>(elem->call_data);
-  calld->pollent_ = pollent;
-}
-
-//
-// CallData implementation
-//
-
-const RetryMethodConfig* RetryFilter::GetRetryPolicy(
```

Comment:
I think this method should stay in this file, since it's part of the `RetryFilter` class, not the call data class.  (I'm not sure why it was in this position in the file to begin with -- it should have been somewhere else.)

Reply:
Yeah agree, this was a copy/paste oversight.